builder/includes/associations.form.inc

```diff
@@ -298,3 +295,159 @@ function xml_form_builder_enable_association($form_name, $id) {
+        '#title' => t('Metadata Datastream ID'),
+        '#required' => TRUE,
+        '#default_value' => $association['dsid'],
+      ),
+      'form_name' => array(
+        '#type' => 'value',
+        '#title' => t('Form Name'),
+        '#value' => $form_name,
+      ),
+      'title_field' => array(
+        '#type' => 'select',
+        '#title' => t('Title Field'),
+        '#description' => t("The form field for the object's label."),
+        '#prefix' => '<div id="ahah-wrapper">',
+        '#suffix' => '</div>',
+        '#options' => $options,
+        '#default_value' => $selected,
+      ),
+      'id' => array(
+        '#type' => 'value',
+        '#value' => $id,
+      ),
+    ),
+  );
+  if (!variable_get('xml_form_builder_use_default_dc_xslts', FALSE)) {
+    $form['fieldset']['transform'] = array(
+      '#type' => 'select',
+      '#title' => t('XSL Transform'),
+      '#options' => xml_form_builder_get_transforms(),
+      '#default_value' => $association['transform'],
+      '#description' => t("An XSL transform for setting the Fedora object's Dublin Core metadata datastream."),
+    );
+    $form['fieldset']['self_transform'] = array(
+      '#type' => 'select',
+      '#title' => t('Self XSL Transform'),
+      '#options' => xml_form_builder_get_self_transforms(),
+      '#default_value' => $association['self_transform'],
+      '#description' => t('An optional transformation applied to form data prior to ingestion.'),
+    );
+  }
+  $form['fieldset']['remove_transform'] = array(
+    '#title' => t('Remove existing template'),
+    '#type' => 'checkbox',
+    '#access' => $association['template'],
+  );
+  $title = $association['template'] ? t('Replace Current Template Document') : t('Upload Template Document');
+  $form['fieldset']['template_file'] = array(
+    '#type' => 'managed_file',
+    '#title' => filter_xss($title),
+    '#upload_location' => 'public://islandora_form_template',
+    '#upload_validators' => array(
+      'file_validate_extensions' => array('xml'),
+    ),
+    '#description' => t('A sample metadata file used to prepopulate the form on ingest.'),
+    '#states' => array(
+      'visible' => array(
+        ':input[name="remove_transform"]' => array('checked' => FALSE),
+      ),
+    ),
+  );
+  $form['fieldset']['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Edit Association'),
+  );
+  return $form;
+}
+
+/**
+ * Default submit handler for the associations edit form.
+ *
+ * @param array $form
+ *   The drupal form.
+ * @param array $form_state
+ *   The drupal form state.
+ */
+function xml_form_builder_edit_association_form_submit(array $form, array &$form_state) {
+  module_load_include('inc', 'xml_form_builder', 'includes/associations');
+  $id = $form_state['values']['id'];
+  $form_name = $form_state['values']['form_name'];
+  $template = '';
+  $file_uploaded = isset($form_state['values']['template_file']);
```

Comment:
This pops warnings as the file is managed the value is a fid. So when the fid isn't 0 it'll have been uploaded.

```

    Notice: Trying to get property of non-object in xml_form_builder_edit_association_form_submit() (line 426 of /var/www/drupal7/sites/all/modules/islandora_xml_forms/builder/includes/associations.form.inc).
    Warning: DOMDocument::load(): Empty string supplied as input in load() (line 426 of /var/www/drupal7/sites/all/modules/islandora_xml_forms/builder/includes/associations.form.inc).
```

Could just do `if ($form_state['values']['template_file'] && !$form_state['values']['remove_transform'])` and remove the `$file_uploaded` var.